krest.py

```diff
@@ -60,21 +74,35 @@ class RetryCfg(object):
         on_other_errors = True
         on_toofast_error = False
 
-    def __init__(self, k2_addr, username, password, ssl_validate=True,
+    def __init__(self, k2_addr, auth: AuthBase=None, username=None, password=None, sdp_id=None,
+                 ssl_validate=True,
                  autodiscover=True,
                  lazy_load_references=True,
                  parse_references=True,
-                 validate_endpoints=True):
+                 validate_endpoints=True,
+                 ):
 
         self.full_endpoint = "%s/%s" % (self.api_prefix, "__full")
 
         self.lazy_load_references = lazy_load_references
         self.parse_references = parse_references
         self.validate_endpoints = validate_endpoints
-
         self.ssl_validate = ssl_validate
+
+        if isinstance(auth, KrestBearerAuth) and not sdp_id:
+            raise Exception("Can't use BearerAuth without sdp_id")
+        if auth and (username or password):
+            raise Exception("Invalid args, Can't use both methods - Auth and username/password")
+
+        self.auth = auth if auth else KrestBasicAuth(username, password)
+
+        if isinstance(auth, KrestBasicAuth) and sdp_id:
+            raise Exception("Invalid args, Can't use both - BasicAuth and sdp_id")
```

Comment:
Keep the validation conditions together
```suggestion
        if isinstance(auth, KrestBasicAuth) and sdp_id:
            raise Exception("Invalid args: Can't use both - BasicAuth and sdp_id")
        if isinstance(auth, KrestBearerAuth) and not sdp_id:
            raise Exception("Can't use BearerAuth without sdp_id")
        if auth and (username or password):
            raise Exception("Invalid args, Can't use both methods - Auth and username/password")

        self.auth = auth if auth else KrestBasicAuth(username, password)
```